httprule/client_test.go

```diff
@@ -117,13 +134,20 @@ func TestErrorStatus(t *testing.T) {
 	require.Equal(t, codes.Unknown, status.Code(err))
 }
 
-func newTestServer(body string, statusCode int) *httptest.Server {
-	h := http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+type testServer struct {
+	*httptest.Server
+	request *http.Request
```

Comment:
This took me a moment to work out - maybe a brief comment like:
"save last request to test server for assertions, e.g. if Header was set correctly "
you will find way better words I know.

Reply:
Will do